addons/base_vat/models/res_partner.py

```diff
@@ -112,37 +116,40 @@ def simple_vat_check(self, country_code, vat_number):
             return bool(self.env['res.country'].search([('code', '=ilike', country_code)]))
         return check_func(vat_number)
 
-    @api.model
-    @tools.ormcache('vat')
-    def _check_vies(self, vat):
-        # Store the VIES result in the cache. In case an exception is raised during the request
-        # (e.g. service unavailable), the fallback on simple_vat_check is not kept in cache.
-        return check_vies(vat)
-
-    @api.model
-    def vies_vat_check(self, country_code, vat_number):
-        try:
-            # Validate against  VAT Information Exchange System (VIES)
-            # see also http://ec.europa.eu/taxation_customs/vies/
-            vies_result = self._check_vies(country_code.upper() + vat_number)
-            return vies_result['valid']
-        except InvalidComponent:
-            return False
-        except Exception:
-            # see http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl
-            # Fault code may contain INVALID_INPUT, SERVICE_UNAVAILABLE, MS_UNAVAILABLE,
-            # TIMEOUT or SERVER_BUSY. There is no way we can validate the input
-            # with VIES if any of these arise, including the first one (it means invalid
-            # country code or empty VAT number), so we return True and ignore the result.
-            _logger.exception("Failed VIES VAT check.")
-            return True
+    @api.depends('vat', 'country_id')
+    def _compute_vies_vat_to_check(self):
+        """ Retrieve the the VAT number, if one such exists, to be used when checking against the VIES system """
+        eu_country_codes = self.env.ref('base.europe').country_ids.mapped('code')
+        for partner in self:
+            if not partner.vat or len(partner.vat) == 1:
+                partner.vies_vat_to_check = ''
+                continue
+            country_code, number = partner._split_vat(partner.vat)
+            if not country_code.isalpha() and partner.country_id:
+                country_code = partner.country_id.code
+                number = partner.vat
+            partner.vies_vat_to_check = (
+                country_code.upper() in eu_country_codes or
+                country_code.lower() in _region_specific_vat_codes
+            ) and self._fix_vat_number(country_code + number, partner.country_id.id) or ''
+
+    @api.depends_context('company')
+    @api.depends('vies_vat_to_check')
+    def _compute_display_vies_check(self):
+        """ Determine whether to show VIES validity on the current VAT number """
+        for partner in self:
+            to_check = partner.vies_vat_to_check
+            if not to_check or to_check[:2].upper() == self.env.company.account_fiscal_country_id.code:
+                partner.display_vies_check = False
+                continue
+            partner.display_vies_check = self.env.company.vat_check_vies
```

Comment:
Only set the value at one place, makes the code more redable
```suggestion
            partner.display_vies_check = (
                to_check
                and to_check[:2].upper() == self.env.company.account_fiscal_country_id.code
                and self.env.company.vat_check_vies
            )
```

But why do you need the check on the country? :thinking: 
This seems enough to me
```suggestion
            partner.display_vies_check = partner.vies_vat_to_check and self.env.company.vat_check_vies
```